### PR TITLE
fixtures: add documentation;

### PR DESCRIPTION
**Pull request by bogdanfinn**

### CODE REVIEW

File: examples/gosoline-fixture-loading/main.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
+//+build fixtures
+
+package main
+
+import (
+	"github.com/applike/gosoline/pkg/application"
+	db_repo "github.com/applike/gosoline/pkg/db-repo"
```
Comment (applike-ss) [inline examples/gosoline-fixture-loading/main.go]: please try not to add these as an alias, they are not required. Also look around in all files that have been touched for unnecessary aliases

File: pkg/fixtures/writer_plain_mysql.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	"github.com/applike/gosoline/pkg/cfg"
+	"github.com/applike/gosoline/pkg/db"
+	"github.com/applike/gosoline/pkg/mon"
+	"reflect"
+)
+
+type PlainMySqlFixtureValue []interface{}
+
+type PlainMySqlFixture struct {
+	TableName string
```
Comment (applike-ss) [inline pkg/fixtures/writer_plain_mysql.go]: usually this kind of metadata we set for the whole fixture writer instance and not per fixture (metadata that applies to all fixtures for this writer)

File: pkg/fixtures/writer_plain_mysql.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	Columns   []string
```
Comment (applike-ss) [inline pkg/fixtures/writer_plain_mysql.go]: should we add these as metadata to the factory instead too? i assume the columns are always the same for each fixture of a table?

File: test/fixtures_mysql_test.go [hunk 4]
```diff
@@ -31,11 +62,11 @@ var TestModelMetadata = db_repo.Metadata{
 	},
 }
 
-func mysqlTestFixtures() []*fixtures.FixtureSet {
+func ormMysqlTestFixtures() []*fixtures.FixtureSet {
```
Comment (applike-ss) [inline test/fixtures_mysql_test.go]: actually i think, i would split the test.
This is due to testing different implementations.
Think of it like: we don't test all implementations in one file, so we should have each test in a different file.

File: docs/fixtures/readme.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	Fixtures []interface{}
+}
+```
+You can easily define multiple fixtures to different destinations in one file and enable or disable them for each FixtureWriter.
+
+* Currently there are 4 different FixtureWriterFactories implemented to load fixtures. 
+    * `DynamoDbFixtureWriterFactory`
+    * `DynamoDbKvStoreFixtureWriterFactory` 
+    * `MysqlOrmFixtureWriterFactory` 
+    * `PlainMySqlFixtureWriterFactory`
```
Comment (applike-ss) [inline docs/fixtures/readme.md]: should we adjust this name to MysqlPlainFixtureWriterFactory?

File: docs/fixtures/readme.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+You can easily define multiple fixtures to different destinations in one file and enable or disable them for each FixtureWriter.
+
+* Currently there are 4 different FixtureWriterFactories implemented to load fixtures. 
+    * `DynamoDbFixtureWriterFactory`
+    * `DynamoDbKvStoreFixtureWriterFactory` 
+    * `MysqlOrmFixtureWriterFactory` 
+    * `PlainMySqlFixtureWriterFactory` 
+
+## Quick Usage
+* During the creation of you Application make sure to pass the `WithFixtures` option and provide fixtures as an argument of type `[]*fixtures.FixtureSet`
```
Comment (applike-ss) [inline docs/fixtures/readme.md]: please correct to `During the creation of your Application...`

File: pkg/fixtures/writer_plain_mysql.go (hunk 2 above)
Comment (bogdanfinn) [reply pkg/fixtures/writer_plain_mysql.go]: Done.

File: pkg/fixtures/writer_plain_mysql.go (hunk 3 above)
Comment (bogdanfinn) [reply pkg/fixtures/writer_plain_mysql.go]: Done.

File: docs/fixtures/readme.md (hunk 5 above)
Comment (bogdanfinn) [reply docs/fixtures/readme.md]: Yeah. true

File: examples/gosoline-fixture-loading/main.go [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+		},
+		{
+			Enabled: true,
+			Writer: fixtures.MySqlPlainFixtureWriterFactory(fixtures.MySqlPlainMetaData{
+				TableName: "plain_fixture_example",
+				Columns:   []string{"id", "name"},
+			}),
+			Fixtures: []interface{}{
+				&fixtures.MySqlPlainFixtureValues{1, "testName1"},
+				&fixtures.MySqlPlainFixtureValues{2, "testName2"},
```
Comment (applike-ss) [inline examples/gosoline-fixture-loading/main.go]: To be consisten, we should write MySql always as Mysql

File: pkg/fixtures/writer_plain_mysql.go [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+		m.logger.Info(fmt.Sprintf("affected rows while fixture loading: %d", ar))
+	}
+
+	m.logger.Infof("loaded %d plain mysql fixtures", len(fs.Fixtures))
+
+	return nil
+}
+
+func (m *mySqlPlainFixtureWriter) buildSql(values *MySqlPlainFixtureValues) (string, []interface{}, error) {
+	insertBuilder := squirrel.Replace(m.metaData.TableName).PlaceholderFormat(squirrel.Question).Columns(m.metaData.Columns...).Values(*values...)
```
Comment (applike-ss) [inline pkg/fixtures/writer_plain_mysql.go]: from the last commits it looks like logic changed a bit, but i guess you checked it still works properly?
I would like to see this line to become a bit shorter, how about this:
```suggestion
	insertBuilder := squirrel.Replace(m.metaData.TableName).
	                          PlaceholderFormat(squirrel.Question).
	                          Columns(m.metaData.Columns...).
	                          Values(*values...)
```

File: test/fixtures_mysql_test.go (hunk 4 above)
Comment (bogdanfinn) [reply test/fixtures_mysql_test.go]: wont do it because of the "boot mysql multiple times during testing"-argument

**Pull request merged**